odoo/tools/speedscope.py

```diff
@@ -31,6 +31,9 @@ def add(self, key, profile):
                 query = entry['query']
                 full_query = entry['full_query']
                 entry['stack'].append((f'sql({shorten(query)})', full_query, None))
+            if 'explain' in entry:
+                explain = entry['explain']
+                entry['stack'].append((f'explain({shorten(explain)})', explain, None))
```

Comment:
I'm not sure this will be really readable (I should test that) but I would suggest create a widget to display that in the ir.profile view form instead of the speedscope representation. It would be easier to use I think.

Reply:
It can be very simple, we just need to parse the json and display it as a list of queries, with the time, and if we have the information, the explain. This way you can just open speedscope, copy the query that was slow and search it in the view form, with a more readable presentation of the query plan and the formated query.